common/src/seg_tree.c

```diff
@@ -100,33 +103,54 @@ get_non_overlapping_range(unsigned long start1, unsigned long end1,
     long start2, long end2, long* new_start, long* new_end)
 {
     if (start1 >= start2 && end1 <= end2) {
-        /* Completely overlapping */
+        /* Segment 2 completely envelops segment 1
+         * nothing left of segment 1 to return
+         * return 1 to indicate this case
+         *
+         *    s1-------e1
+         * s2-------------e2
+         */
         return 1;
     } else if (start1 < start2) {
-        /*
-         * s1 ------- e1
-         *      s2--------e2
-         *    ---- non-overlap
+        /* Segment 1 inlcudes a portion before segment 2 starts
+         * return start/end of that leading portion of segment 1
+         *
+         * s1-------e1
+         *     s2--------e2
+         *   ---- non-overlap
          *
          * also:
          *
-         * s1 -------------------e1
-         *      s2--------e2
-         *    ---- non-overlap
+         * s1-------------------e1
+         *     s2--------e2
+         *   ---- non-overlap
          */
         *new_start = start1;
-        *new_end = MIN(end1, start2 - 1);
+        *new_end   = MIN(end1, start2 - 1);
     } else if (start1 > start2 && end1 > end2) {
-        /*
-         *       s1 ----- e1
-         *  s2------- e2
+        /* Segment 1 extends past end of segment 2
+         * return start/end of trailing portion of segment 1
+         *
+         *       s1-----e1
+         *  s2-------e2
+         *           --- non-overlap
          */
         *new_start = MAX(start1, end2 + 1);
-        *new_end = end1;
+        *new_end   = end1;
     } else if (start1 == start2 && end1 > end2) {
+        /* Segment 1 extends past end of segment 2
+         * return start/end of trailing portion of segment 1
+         *
+         *  s1----------e1
+         *  s2-------e2
+         *           --- non-overlap
+         */
         *new_start = end2 + 1;
-        *new_end = end1;
+        *new_end   = end1;
     }
```

Comment:
Can we combine this branch (start1 == start2 && end1 > end2) with the branch immediately above (start1 > start2 && end1 > end2)?  I think they should both be returning the same trailing portion of segment 1 right?

Maybe just simplify the check to (end2 > end1) since the branches before these two handle any case when (start1 < start2), so that we know (start1 >= start2) by this point.